clippy_lints/src/let_if_seq.rs

```diff
@@ -67,7 +67,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for LetIfSeq {
                 if let Some(expr) = it.peek();
                 if let hir::StmtDecl(ref decl, _) = stmt.node;
                 if let hir::DeclLocal(ref decl) = decl.node;
-                if let hir::PatKind::Binding(mode, canonical_id, ref name, None) = decl.pat.node;
+                if let hir::PatKind::Binding(mode, canonical_id, ref ident, None) = decl.pat.node;
```

Comment:
`ident` is small and `Copy`, so the `ref` is not necessary (applicable to all `PatKind::Binding`s).

Reply:
I thought there is clippy lint for it, nice catch.

Reply:
I think the lint for that applies only on function arguments and not on matching. This could be a nice addition to that lint though!